### PR TITLE
✨ Add support for compact Snackbar

### LINKED ISSUE #1756: Add compact density for Snackbar
- [x] React

- [ ] Design

- [ ] Documentation

### PR DESCRIPTION
**Pull request by laliq**
Subtask of #1756

### CODE REVIEW

File: libraries/core-react/src/components/Snackbar/Snackbar.tokens.ts [hunk 1]
```diff
@@ -57,4 +58,9 @@ export const snackbar: Snackbar = {
     ...typography,
     color,
   },
+  modes: {
+    compact: {
+      height: shape._modes.compact.straight.minHeight,
```
Comment (mimarz) [inline libraries/core-react/src/components/Snackbar/Snackbar.tokens.ts:63]: Looks like this should be `minHeight` as its what used in the "default".

**Review by mimarz — CHANGES_REQUESTED** [review]
The default value for `minHeight` should probably be changed to the [correct shape in Figma](https://www.figma.com/file/0bGXR2sCwMVSDNyyzu5BXrO5/UI%E2%80%94User-Interface?node-id=17659%3A40216)

Looks like you missed to updated `SnackbarActions` styled component with new theme syntax.

**Review by oddvernes — APPROVED** [review]
LGTM 👍

File: libraries/core-react/src/components/Snackbar/Snackbar.stories.tsx [hunk 2]
```diff
@@ -67,3 +68,23 @@ export const WithAction: Story<SnackbarProps> = () => {
 }
 
 WithAction.storyName = 'With action'
+
+export const Compact: Story<SnackbarProps> = () => {
+  const [open, setOpen] = useState(false)
+  const [density] = useState<Density>('comfortable')
```
Comment (mimarz) [inline libraries/core-react/src/components/Snackbar/Snackbar.stories.tsx:74]: Should probably be `compact` here aswell?

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**